README.md

```diff
@@ -345,6 +345,24 @@ included by default unless you choose which one to exclude with the
 then you have to specify each backend that you want individually. You can get
 the full list of these flags with the `--help` flag.
 
+#### --auto-config-uri
+This is a special URI which if set, will try and pull a config from that
+location on startup. It will use the cookie file stored at
+`--auto-config-cookie-path` if specified. If successful, it will check if the
+config is different from what is currently stored. If so then it will validate
+if it is a valid json config. If so it will replace (overwrite!) the current
+config and then run with that!
+
+For example: `--auto-config-uri 'https://example.com/config.json'`.
+
+#### --auto-config-cookie-path
+This is a special URI which if set will point to a netscape/libcurl style cookie
+file to use when making the get download requests. This is useful if you store
+your config behind some gateway that needs a magic cookie for auth. It accepts
+the tilde (`~`) character to use for `$HOME` directory path expansion.
+
+For example:  `--auto-config-cookie-path '~/.secret/cookie'`.
```

Comment:
I'm not sure what you mean here. The expectation is that some other tool (for example your internal systems single-sign-on system) would write out this file automatically at desktop unlock. And this just tells `yesiscan` where to look to read those credentials. Please comment again if there's something I missed here.

Reply:
> I'm not sure what you mean here.

If you're not sure what I mean, don't click the "Resolve conversation" button!

cURL has both a `--cookies` option and a `--cookie-jar` option. The former is read-only for supplying cookies to the program, the latter is for storing cookies obtained by the program.

When you authenticate to a web service you get redirected around (as much as 20 times, haha) to collect cookies so that ultimately you can go directly to the original URL you were trying to reach.

If the tool can store those cookies (not only read them) then the next time you run the tool it can go straight there (unless the cookies are expired, or the session is expired on the server side).

I think it's a fine choice to only support a read-only cookie file that's created by some other tool, and not store any additional cookies collected on the way, but if we're making that choice it's good to make it explicit that `yesiscan` will not write to that file (especially given that `yesiscan` might unexpectedly write to other files you've provided in its options).

Does that clarify?

Reply:
> cURL has both a --cookies option and a --cookie-jar option. The former is read-only for supplying cookies to the program, the latter is for storing cookies obtained by the program.

Correct-- since the complexities of handing every different auth system (including undocumented unknown ones) is outside of the scope of this system, we don't try to. As a result, we're only interested in reading any existing cookies that other tools have create for us. Most third-party tools create such cookies and store them in a well-known path. In yesiscan we are asking the user to provide that path for us. We won't ever write to it.

I have added a doc addition in the subsequent patch to clarify that we are only reading from this path and never writing. As a result I will resolve this thread. If you'd like to continue the discussion, please add a comment which will automatically unresolve it.

Reply:
> If you'd like to continue the discussion, please add a comment which will automatically unresolve it.

Adding this comment did not unresolve the conversation.

Reply:
> since the complexities of handing every different auth system (including undocumented unknown ones) is outside of the scope of this system, we don't try to

There's no complexity that `yesiscan` would need to know about implied here. Browsers don't need to know any of that complexity either (and they don't, modulo U2F/FIDO because those require hardware). The complexity is hidden behind the fact that the auth system requires and gives cookies. So to support auth, you only need to handle cookies correctly. And indeed `yesiscan` is handling those cookies, but only in memory.

I think it's acceptable for `yesiscan` to not be able to persist cookies, but it does have a cost in startup time, bandwidth, and load on services that should be understood and documented, or considered and implemented.